client/post-editor/post-editor.jsx

```diff
@@ -592,10 +602,13 @@ export const PostEditor = React.createClass( {
 
 		previewPost = function() {
 			if ( this._previewWindow ) {
-				this._previewWindow.location = this.state.previewUrl;
+				this._previewWindow.location = ( this.state.previewAction === 'view' ? this.state.post.URL : this.state.previewUrl );
 				this._previewWindow.focus();
 			} else {
-				this._previewWindow = window.open( this.state.previewUrl, 'WordPress.com Post Preview' );
+				this._previewWindow = window.open(
+					( this.state.previewAction === 'view' ? this.state.post.URL : this.state.previewUrl ),
```

Comment:
Good call. Ternary converted to helper function. `getPreviewUrl` introduces a tiny bit of shadowing since `PostEditStore` has [a method with the same name](https://github.com/Automattic/wp-calypso/blob/d008cf71b2bdce6304b572c6236b20589b120d75/client/post-editor/post-editor.jsx#L101). This naming makes the most sense and hopefully it's still straightforward enough.